src/coprocessor/dag/rpn_expr/impl_time.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use cop_codegen::rpn_fn;
+
+use super::super::expr::EvalContext;
+
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::Error;
+use crate::coprocessor::Result;
+
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn date_format(
+    ctx: &mut EvalContext,
+    t: &Option<DateTime>,
+    layout: &Option<Bytes>,
+) -> Result<Option<Bytes>> {
+    use std::str::from_utf8;
+
+    if t.is_none() || layout.is_none() {
+        return Ok(None);
+    }
+    let (t, layout) = (t.as_ref().unwrap(), layout.as_ref().unwrap());
```

Comment:
What will `mysql` do when `t` is an invalid zero and layout is none?

Reply:
> What will `mysql` do when `t` is an invalid zero and layout is none?

It's layout is none both in and not in strict mode.

Reply:
```
mysql> select date_format("0000-00-00","%Y");
+--------------------------------+
| date_format("0000-00-00","%Y") |
+--------------------------------+
| NULL                           |
+--------------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+----------------------------------------+
| Level   | Code | Message                                |
+---------+------+----------------------------------------+
| Warning | 1292 | Incorrect datetime value: '0000-00-00' |
+---------+------+----------------------------------------+
1 row in set (0.00 sec)

mysql> select  @@session.sql_mode;
+-----------------------------------------------------------------------------------------------------------------------+
| @@session.sql_mode                                                                                                    |
+-----------------------------------------------------------------------------------------------------------------------+
| ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION |
+-----------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
Thanks for your review. On account of the issue https://github.com/tikv/tikv/issues/3953 , test for datetime '0000-00-00' will not be test now.

As for the warnings, if `t.invalid_zero()` works well, `EvalContext.handle_invalid_time_error` will be called, which will check the `sql_mode` of the system and show warning if necessary.